### PR TITLE
chore: refactor scripts for octokit update 

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Update our version of octokit to latest and refactor release scripts accordingly.

/cc @BinaryMuse @jkleinsc 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: script/release.js [hunk 1]
```diff
@@ -195,37 +192,41 @@ async function createReleaseShasums (release) {
...
 }
 
-async function uploadShasumFile (filePath, fileName, release) {
-  const githubOpts = {
-    owner: 'electron',
-    repo: targetRepo,
-    id: release.id,
-    filePath,
+async function uploadShasumFile (filePath, fileName, releaseId) {
+  const uploadUrl = `https://uploads.github.com/repos/electron/${targetRepo}/releases/${releaseId}/assets{?name,label}`
```
Comment (jkleinsc) [inline script/release.js]: Why do we need a upload url here?

File: script/release.js [hunk 2]
```diff
@@ -162,12 +163,8 @@ function checkVersion () {
 
 function runScript (scriptName, scriptArgs, cwd) {
   const scriptCommand = `${scriptName} ${scriptArgs.join(' ')}`
-  const scriptOptions = {
-    encoding: 'UTF-8'
-  }
-  if (cwd) {
-    scriptOptions.cwd = cwd
-  }
+  const scriptOptions = { encoding: 'UTF-8' }
```
Comment (jkleinsc) [inline script/release.js]: This refactor is unneeded

File: script/release.js (hunk 1 above)
Comment (codebytere) [reply script/release.js]: https://octokit.github.io/rest.js/#api-Repos-uploadReleaseAsset it's required now

File: script/upload-to-github.js [hunk 3]
```diff
@@ -1,48 +1,68 @@
...
+  }
 
-const githubOpts = {
-  owner: 'electron',
-  repo: targetRepo,
-  id: releaseId,
-  filePath: filePath,
-  name: fileName
+  return {
+    'content-type': options[extension],
```
Comment (BinaryMuse) [inline script/upload-to-github.js]: How do you feel about a safe fallback here?

```suggestion
    'content-type': options[extension] || 'application/octet-stream',
```

**Pull request merged**